mlserver/codecs/utils.py

```diff
@@ -16,19 +17,19 @@
 )
 
 Parametrised = Union[InferenceRequest, RequestInput]
-Tagged = Union[MetadataTensor]
+Tagged = Union[MetadataTensor, ModelSettings]
 DecodedParameterName = "_decoded_payload"
 
 
 def _get_content_type(
-    parametrised_obj: Parametrised, tagged_obj: Optional[Tagged] = None
+    request: Parametrised, metadata: Optional[Tagged] = None
 ) -> Optional[str]:
-    if parametrised_obj.parameters and parametrised_obj.parameters.content_type:
-        return parametrised_obj.parameters.content_type
+    if request.parameters and request.parameters.content_type:
+        return request.parameters.content_type
 
-    if tagged_obj is not None:
-        if tagged_obj.tags and tagged_obj.tags.content_type:
-            return tagged_obj.tags.content_type
+    if metadata is not None:
```

Comment:
do we care if `request` and `metadata` disagree on `content_type` ?

Reply:
In that case, I think that the content type `request` should override whatever it's on the metadata one (mainly because in that case the user would be explicitly saying what content type they want to set).